tcxparser/tcxparser.py

```diff
@@ -98,6 +98,39 @@ def hr_min(self):
         """Minimum heart rate of the workout"""
         return min(self.hr_values())
 
+    def hr_percent_in_zones(self, zones):
+        """Percentage of workout spent in eacht heart rate zone.
+
+        Given these user's heart rate zones:
+        {
+            "Z0": (0, 119),
+            "Z1": (120, 199),
+            "Z2": (200, 240),
+        }
+
+        We return something like:
+        {
+            "Z0": 5,
+            "Z1": 95,
+            "Z2": 0,
+        }
+
+        Correct calculation relies on evenly spaced measurement times.
+        """
+        # count number of HR measurements per zone
+        per_zone = dict(zip(zones.keys(), [0 for x in zones.keys()]))
+        hr_values = self.hr_values()
+        for hr in hr_values:
+            for zone_name, zone_boundaries in zones.items():
+                if hr >= zone_boundaries[0] and hr <= zone_boundaries[1]:
+                    per_zone[zone_name] += 1
+
+        # convert counts to percentages
+        nr_hr_values = len(hr_values)
+        for name, count in per_zone.items():
+            per_zone[name] = round(100 * count / nr_hr_values)
```

Comment:
I'm not 100% sure, but I think it would be possible for `nr_hr_values` to be zero so we'd get a division by zero error here (like if a user called this method without having any valid HR data parsed in their data file). So we should probably just make sure that `nr_hr_values` is not zero. Better safe than sorry :)

Reply:
I agree. As errors should not go unnoticed, i decided to raise a NoHeartRateDataError in this case.